core/read.c

```diff
@@ -2690,6 +2690,68 @@ static int isom_read_skip_box_extra_bytes( lsmash_file_t *file, isom_box_t *box,
     return 0;
 }
 
+#define READ_STRING( box_name, field )                                            \
+    do                                                                            \
+    {                                                                             \
+        uint64_t pos = lsmash_bs_count( bs );                                     \
+        box_name->field ## _length = box->size - pos;                             \
```

Comment:
Won't this overallocate for any mid-box string?

Reply:
Yep, it will. Do you want me to shrink it down to the final size with realloc at the end, or do you have other preferred ways of doing it?